Add basic layers samples

fe1ixz opened the pull request:
This PR adds a set of basic WebXR Layers samples, there will be another PR that adds the more advanced samples. @toji

toji commented:
Thank you again for breaking these up into smaller PRs! It makes them much easier to review comprehensively. Before looking into the code much I checked this PR out locally and gave the samples a run on my Quest 2. For the most part they look great! The layer rendering is extremely crisp. There's a few questions or points of confusion that I think should be addressed before landing them, though:

 - You can remove the "Does my browser support WebXR?" section from the top of the page. We only really need that on the main page.
 - I was a bit confused at first by the projection layer sample, because I kept waiting for something to load. Eventually I realized that the fact that I could see the FPS counter meant that it was working, and that sample had no environment mesh. But for several minutes I thought it was broken. I suggest having that sample load one of the environments we have from the other samples just to avoid users thinking something is wrong. Since you're using a 'local' reference space the space scene is probably a good one.
 - The other samples don't need an environment, as the other layers give you an indication that the sample worked.
 - The stereo image on the quad layer and cylinder layer sample appears to have the left/right views inverted. It's hard to tell. Also, that image (of the bird from Big Buck Bunny) is really blurry, and looks like a screencap of the video. It would be preferable if we could find a higher quality stereo image to demo, since the whole point if this API is improving image quality.
 - The images with the robots are really nice looking, but I want to ensure we have the rights to use them as part of these samples. If they are creatives commons licensed, as are most of the other assets these samples use, then we need to include acknowledgements of where the images were obtained from. Same goes for the equirect test patterns (and if you created them yourself please give yourself credit!) If the images are not under an appropriately open license please find some that we can use freely.
 - The image of the chessboard used for the stereo equirect sample has the left/right views inverted.
 - If I set the radius for the equirect sample to something like 5 or 1 nothing renders. Also, it's difficult to see the effect that value has if it can't be adjusted in realtime, so either finding a way to visualize it better or hiding that option for now may be appropriate.

fe1ixz commented:
> Thank you again for breaking these up into smaller PRs! It makes them much easier to review comprehensively. Before looking into the code much I checked this PR out locally and gave the samples a run on my Quest 2. For the most part they look great! The layer rendering is extremely crisp. There's a few questions or points of confusion that I think should be addressed before landing them, though:
> 
> * You can remove the "Does my browser support WebXR?" section from the top of the page. We only really need that on the main page.
> * I was a bit confused at first by the projection layer sample, because I kept waiting for something to load. Eventually I realized that the fact that I could see the FPS counter meant that it was working, and that sample had no environment mesh. But for several minutes I thought it was broken. I suggest having that sample load one of the environments we have from the other samples just to avoid users thinking something is wrong. Since you're using a 'local' reference space the space scene is probably a good one.
> * The other samples don't need an environment, as the other layers give you an indication that the sample worked.
> * The stereo image on the quad layer and cylinder layer sample appears to have the left/right views inverted. It's hard to tell. Also, that image (of the bird from Big Buck Bunny) is really blurry, and looks like a screencap of the video. It would be preferable if we could find a higher quality stereo image to demo, since the whole point if this API is improving image quality.
> * The images with the robots are really nice looking, but I want to ensure we have the rights to use them as part of these samples. If they are creatives commons licensed, as are most of the other assets these samples use, then we need to include acknowledgements of where the images were obtained from. Same goes for the equirect test patterns (and if you created them yourself please give yourself credit!) If the images are not under an appropriately open license please find some that we can use freely.
> * The image of the chessboard used for the stereo equirect sample has the left/right views inverted.
> * If I set the radius for the equirect sample to something like 5 or 1 nothing renders. Also, it's difficult to see the effect that value has if it can't be adjusted in realtime, so either finding a way to visualize it better or hiding that option for now may be appropriate.

I've made changes to:
1. Remove WebXR compatibility check from layers sample page
2. Remove equirect radius option
3. Add space environment mesh to projection layer sample

The robot image was added a while ago by a former member of our team, we cannot find the rights information on these images any more, we will try to find images to replace these, as well as the bird image.

fe1ixz commented:
@toji I have:
1. replaced the cubemap images and added photo credit
2. replaced the rectangular stereo/mono images used in cylinder and quad layer sample and added photo credit
3. corrected the invert stereo image chess-pano-4k and chess-pano-4k180

This should address all the issues you raised in your original feedback comment, please let me know if there is more to be done. Thanks!

toji commented on layers-samples/index.html:
```diff
@@ -0,0 +1,216 @@
+          links.classList.add('links');
+
+          let sourceLink = document.createElement('a');
+          sourceLink.href = 'https://github.com/fe1ixz/webxr-samples/blob/master/layers-samples/' + page.path;
```
```suggestion
          sourceLink.href = 'https://github.com/immersive-web/webxr-samples/blob/master/layers-samples/' + page.path;
```

toji commented on layers-samples/proj-layer.html:
```diff
@@ -0,0 +1,258 @@
+
+        // Note that for the complexity of the scene in this samples this
+        // won't make much visible performance difference, but we're using the
+        // more efficient pattern anyway as a way of promoting best practices.
```
This entire comment block is unnecessary for this sample, as it's covered in a more focused manner in a separate sample. Let's remove it so that readers of this code can stay focused on the projection layer aspects.

toji commented on layers-samples/cube-layer.html:
```diff
@@ -0,0 +1,303 @@
+        only needs to be drawn once or rarely need refreshing. The
+        same thing can be accomplished with projection layers, but
+        since projection layers redraws all objects every frame, cube
+        layer is a much more efficient alternative.
```
This can be addressed in a follow up, but it appears that this sample displays the cube maps backwards? I would never have noticed if not for the fact that the text on the arcade cabinet in the mono image is reversed. On close inspection it looks like the same flipping is happening with the stereo layer, though the stereo effect is correct.

fe1ixz replied:
Interesting! Let me investigate this a bit and address in a follow up PR.

toji commented on layers-samples/index.html:
```diff
@@ -0,0 +1,216 @@
+    <header class='header'>
+      <div id='nav'>
+        <a href='../'>Samples</a>
+        <a href='./' class='selected'>Layers Samples</a>
```
```suggestion
        <a href='./' class='selected'>Layers Samples</a>
        <a href='proposals/'>Proposals</a>
```

Just for consistency

toji commented on tests/index.html:
```diff
@@ -90,6 +90,7 @@
         <div id='nav'>
             <a href='../'>Samples</a>
+            <a href='../layers-samples/'>Layers Samples</a>
             <!--a href='../proposals/'>Proposals</a-->
```
```suggestion
            <a href='../proposals/'>Proposals</a>
```

I have no idea why this was commented out before. It's not a problem you introduced, but while you're touching this file it would be nice to fix it.

toji reviewed (commented):
The new images and tweaks to the samples are looking great! Thank you so much! There's a few very minor changes that I think we should make before landing, but I'm pretty happy with where these are at.

fe1ixz commented:
@toji Thanks for the feedback, I've made changes for all of these issues except for the inverted issue, if you feel that this is not a significant blocker, we can land this first and I will investigate and address this in a follow up PR.

toji commented:
Excellent! Thank you!

If you can figure out how to flip the cube maps in a follow-up patch it would be appreciated, and I'll let you know if anything else comes up. This is a really fantastic addition to the samples page, and I really appreciate your patience and perseverance in getting them landed!

fe1ixz commented:
> Excellent! Thank you!
> 
> If you can figure out how to flip the cube maps in a follow-up patch it would be appreciated, and I'll let you know if anything else comes up. This is a really fantastic addition to the samples page, and I really appreciate your patience and perseverance in getting them landed!

And thank you very much for bearing with me! This is a lot of code to review and I really appreciate you taking the time! Super excited to land all the samples and hopefully get more people onboard!
